Checking in modified version of the installer (#498)
grafana/agent: docs/windows.md

> # Windows Installation

Reviewer: Can you link to this in `docs/getting-started.md`, add it to the table of contents in `docs/README.md`, and add another reference for it in `production/README.md`?

## Overview

The installer will install Grafana Agent into the default directory `C:\Program Files (x86)\Grafana Agent`. [Windows Exporter](https://github.com/grafana/windows_exporter) can be enabled with all default options. 

## Installation

After installation ensure that you can reach `http://localhost:12345/-/healthy` and `http://localhost:12345/agent/api/v1/targets`. 

If Grafana Agent is re-installed and an agent-config.yaml already exists it will not overwrite the existing one.

> After installation adjust agent-config.yaml to meet specific needs and restart the agent to see changes.

Reviewer: ```suggestion
After installation, you can adjust `C:\Program Files (x86)\Grafana Agent\agent-config.yaml` to meet your specific needs. After changing the configuration file, the Grafana Agent service must be restarted to load changes to the configuration.
```

> ## Silent Installation

Reviewer: ```suggestion
### Silent Installation
```

Silent installation can be achieved via  `grafana-agent-installer.exe /S  /EnableExporter "true"`. EnableExporter enables or disables Windows Exporter, default is `false`.

## Security

> The Agent configuration is installed alongside the Agent itself, by default. Depending on your configuration, you may not want that for security reasons and may instead want to make it protected. The configuration is by default stored in `C:\Program Files (x86)\Grafana Agent`. You can change the configuration location by running `sc config "Grafana Agent" binpath= "<installed_directory>\agent-windows-amd64.exe -config.file=\"<new_path>\agent-config.yaml\""` in cmd as an admin.

Reviewer: ``````suggestion
A configuration file for the Agent is provided by default at `C:\Program Files (x86)\Grafana Agent`. Depending on your configuration, you may wish to modify the default permissions of the file or move it to another directory. 

When changing the location of the configuration file, you must update the Grafana Agent service to load the new path. Run the following in an elevated prompt, replacing `<new_path>` with the full path holding `agent-config.yaml`:

```
sc config "Grafana Agent" binpath= "<installed_directory>\agent-windows-amd64.exe -config.file=\"<new_path>\agent-config.yaml\""
```
``````

## Uninstall

> Via Remove Programs or uninstaller.exe in the directory the Agent is installed in. This will turn off and remove the Agent then delete any installed files in the applications directory.

Reviewer: ```suggestion
If the Grafana Agent is installed using the installer, it can be uninstalled via Windows' Remove Programs or `C:\Program Files (x86)\Grafana Agent\uninstaller.exe`. Uninstalling the Agent will stop the service and remove it from disk. This will include any configuration files in the installation directory. 
```

## Logs

> Logs are currently not stored anywhere for the services version of Agent, in the future logs will be available via Event Viewer.

Reviewer: ```suggestion
Logs from the Grafana Agent service are currently not stored anywhere. In the future version, logs will be written to Event Viewer.
```

# Windows Installation

## Overview

The installer will install Grafana Agent into the default directory `C:\Program Files (x86)\Grafana Agent`, optionally you can select to enable the [Windows Exporter](https://github.com/grafana/windows_exporter) with all default options. 

## Installation

> ![](./assets/remote_options.png)

Author: Sure! Does mean we lose the ability to easily spin up one pointing somewhere via the silent install option but users can always have a post-install script for that.

> ![](./assets/remote_options.png)

Reviewer: right, I think in general most users are going to need to modify the config file after install _anyway_ since the defaults don't collect very much. Not a huge loss, but we can always backpedal on this
